src/library/simplified/extension/checkin-date.js

```diff
@@ -5,6 +5,10 @@ if (utag.isHSR() && b['entity.hotels.search.hotelParameters.checkInDate'] != und
 {
     b['checkInDate'] = b['entity.hotels.search.hotelParameters.checkInDate'];
 }
+else if (utag.isRailSearchResults() && b['entity.railSearch.searchParameters.journeyLegs.0']!= undefined )
```

Comment:
not necessary to add a second check, just check it on line 8 is good enough, then assign it to checkInDate inside the block.